adam/language_specific/chinese/chinese_language_generator.py

```diff
@@ -266,24 +646,57 @@ def add_classifier(
             if len(possession_relations) > 1:
                 raise RuntimeError("Cannot handle multiple possession relations")
             elif len(possession_relations) == 1:
-                # handle the possession relation if there is one. We don't need to case on person in Chinese
-                # since all possessives are expressed NP+de+NP
-                possessor = self._noun_for_object(possession_relations[0].first_slot)
+                # handle the possession relation if there is one
+                # TODO: since we can't distinguish nodes of the same ontology type right now (https://github.com/isi-vista/adam/issues/55),
+                #  we must manually fill in wo and ni to prevent "de" from attaching to both in a case of a sentence like
+                #  "I have my ball" (which would turn into wo de you wo de qiu rather than wo you wo de qiu)
+                possessor = None
+                if IS_SPEAKER in possession_relations[0].first_slot.properties:
+                    possessor = DependencyTreeToken("wo3", NOUN)
+                elif IS_ADDRESSEE in possession_relations[0].first_slot.properties:
+                    possessor = DependencyTreeToken("ni3", NOUN)
+                # if the possessor is a 3rd person, check that "has" isn't the main verb
+                elif (
+                    IGNORE_HAS_AS_VERB not in self.situation.syntax_hints
+                    and not self.situation.is_dynamic
+                ):
+                    return
+                    # TODO: we currently return here since we can't handle one possessive node and one not -- once we fix this we don't need this case
```

Comment:
Should this perhaps error out rather than silently return? If in the future we go to implement needing this feature we get an error rather than a silent failure. Silent failures are significantly hard to debug. We should also track this in an issue.

It's also possible I'm not understanding the case in which we simply return here, in that case the comment needs to just be clearer.

Reply:
@LichMaster98 I updated the comment so it should be clearer now; this is in reference to issue #55 which also affects the English generator and this is how it's handled in the English one. I tend to think "Dad has a cookie" is better than just throwing an error in these cases but I can change it if you prefer.

Reply:
Ok! If it's handled this way in the English language generator that's fine then.